modules/storage_account/storage_account.tf

```
@@ -14,20 +14,20 @@ resource "azurecaf_name" "stg" {
   use_slug      = var.global_settings.use_slug
 }
 
-# Tested with :  AzureRM version 2.61.0
+# Tested with :  AzureRM version 3.45
 # Ref : https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account
 
-resource "azurerm_storage_account" "stg" {
+resource "azurerm_storage_account" "stg" {  
   name                              = azurecaf_name.stg.result
   account_tier                      = try(var.storage_account.account_tier, "Standard")
   account_replication_type          = try(var.storage_account.account_replication_type, "LRS")
   account_kind                      = try(var.storage_account.account_kind, "StorageV2")
   access_tier                       = try(var.storage_account.access_tier, "Hot")
   allow_nested_items_to_be_public   = try(var.storage_account.allow_nested_items_to_be_public, var.storage_account.allow_blob_public_access, false)
-  cross_tenant_replication_enabled  = try(var.storage_account.cross_tenant_replication_enabled, null)
+  cross_tenant_replication_enabled  = try(var.storage_account.cross_tenant_replication_enabled, true)
   edge_zone                         = try(var.storage_account.edge_zone, null)
   enable_https_traffic_only         = try(var.storage_account.enable_https_traffic_only, true)
-  infrastructure_encryption_enabled = try(var.storage_account.infrastructure_encryption_enabled, null)
+  infrastructure_encryption_enabled = try(var.storage_account.infrastructure_encryption_enabled, false)
```

Prefer to defer to provider defaults 

```suggestion
  infrastructure_encryption_enabled = try(var.storage_account.infrastructure_encryption_enabled, null)
```